text/0000-volta-run.md

```diff
@@ -59,14 +85,43 @@ There are alternative names we could use for the command, such as:
+Instead of having `--node <version>` and `--yarn <version>`, we could support a more general syntax of `--with <tool>@<version>`. That would align with the syntax that `volta install` and `volta pin` use. However, it adds some potential ambiguity in the command:
+
+- When a user runs `volta pin yarn`, that is interpreted as `volta pin yarn@latest`.
+- If a user runs `volta run --with yarn`, should that _also_ be interpreted as `volta run --with yarn@latest`? It's not clear that's necessarily the best default, or that any default makes sense.
+- Should we instead require the `@<version>` in `volta run`? If so, then the syntax is different from `volta install`, but in a subtle and somewhat hard-to-understand way.
+
+Beyond the potential confusion, there is also the issue that for common use-cases, `--with node@10 --with yarn@1.17` is more verbose than `--node 10 --yarn 1.17`. Given that `volta run` will often be used for one-off commands, making the syntax shorter is a boon to usability.
+
 ## Platform Resolution
 
-Instead of inheriting the platform from the current, we could treat it as purely defined on the command-line. This would give users total control over all of the pieces of the platform, including allowing them to remove `yarn`, even if it is specified in the project or default platforms.
+Instead of inheriting the platform from the current, we could treat it as purely defined on the command-line. However, this would mean that if a user doesn't specify `--node`, no commands will work, since all JS tools ultimately require `node` at some point. Additionally, users have to specify `yarn` if their command needs it, even if they only want to change the version of `node` being used.
+
+## Overriding the shims directly
+
+Instead of (or in addition to) the `volta run` command, we could provide the ability for users to override the shims using environment variables (e.g. `VOLTA_NODE` to change the version of Node). This would provide a different way of specifying the platform that overrides the normal resolution algorithm. However, ultimately this approach has a few issues that clash with the goals of Volta:
+
+- It requires users to think about the existence of Volta when running the shims. When a user calls `yarn`, the goal of Volta is that they shouldn't have to think about the fact that Volta is figuring it out, it should just work as they expect. If we add environment variables that can change how those work, it could lead to difficult-to-debug situations where a user has an exported variable in their terminal that is breaking things and it's not immediately obvious why.
+
+- It's not as ergonomic cross-platform. As mentioned above, Windows doesn't have as easy as shorthand for setting environment variables on a given command, so Windows users would have to jump through more hoops to get access to this behavior. Volta is committed to keeping Windows as a first-class citizen, so we don't want to break that by relying too much on Unix-isms.
```

Comment:
IMHO, this is the deal breaker for this one.

Reply:
Agreed, especially as we’re already supporting the `--env` option in the other use-case.